go/vt/sqlparser/sql.y

```diff
@@ -1290,6 +1290,13 @@ alter_statement:
   {
     $$ = &DDL{Action: AlterStr, Table: $4}
   }
+| ALTER ignore_opt TABLE table_name ADD column_opt column_definition skip_to_end
```

Comment:
The new rule to support `ADD my_column`...

Reply:
Would move this rule above the ignored_alter_object_type one above it. The order can matter in some grammars (if there is any ambiguity, which this one has).

Alternately, you might also try to introduce an open_paren_opt and close_parent_opt into the main add column rule to reduce duplication.

Reply:
Updated. I actually placed it there because I thought there might be ambiguity and I wanted explicitly ignored alter types to hit the ignore rule; but it turns out column definition starts with `ID` and `ID` is not explicitly ignored after your work, so moving it up works too.

I didn't merge the rules because eventually we will want to support `ALTER TABLE my_table ADD COLUMN (one_column int, two_columns varchar(10))` which has no analog in the non-parentheses case.